include/dxc/Support/dxcapi.use.h

```diff
@@ -13,27 +13,40 @@
 #define __DXCAPI_USE_H__
 
 #include "dxc/dxcapi.h"
-#include "llvm/Support/DynamicLibrary.h"
 
 namespace dxc {
 
 // Helper class to dynamically load the dxcompiler or a compatible libraries.
 class DxcDllSupport {
 protected:
-  using DynamicLibrary = llvm::sys::DynamicLibrary;
-  DynamicLibrary m_dll;
+  HMODULE m_dll;
   DxcCreateInstanceProc m_createFn;
   DxcCreateInstance2Proc m_createFn2;
 
+  #ifndef _WIN32
+  void FreeLibrary(void* handle) {
```

Comment:
It's a little odd to have methods matching the name of win32 functions under #ifndef for other platforms (plus they could be static).  Might look cleaner if you had static methods that don't collide in name with win32 functions, and put the #ifdef code within each method body.